syntaxes/vscode-scl.tmLanguage.json

```diff
@@ -91,15 +97,28 @@
 					"name": "storage.type.function.scl"
 				},
 				{
-					"match": "(\"(\\w+?)\")",
+					"match": "(\"(\\w+?)\")|\\#.*\\[\\d\\]|\\#\\w*(?=\\()",
 					"name": "entity.name.function.scl"					
+				},
+				{
+					"captures": {
+						"1": {"name": "storage.type.function.scl"},
+						"2": {"name": "entity.name.function.scl"},
+						"3": {"name": "entity.name.function.scl"}
+					},
+					"match": "(?i)\\b(data_block)\\s+(\"(\\w*)\")?",
+					"name": "storage.type.function.scl"
+				},
+				{
+					"match": "(?i)\\b(END_DATA_BLOCK)\\b",
+					"name": "storage.type.function.scl"
 				}
 			]	
 		},
 		"support":{
 			"patterns": [
 				{
-					"match": "(?i)\\b(ANY|ARRAY|BOOL|BYTE|CHAR|COUNTER|DATE|DATE_AND_TIME|SINT|SINT|INT|DINT|LINT|DT|DWORD|POINTER|REAL|LREAL|S5TIME|STRING|TIME|TIMER|TIME_OF_DAY|TOD|USINT|UINT|UDINT|ULINT|VOID|WORD|WSTRING)\\b",
+					"match": "(?i)\\b(ANY|ARRAY|BOOL|BYTE|CHAR|COUNTER|DATE|DATE_AND_TIME|SINT|SINT|INT|DINT|LINT|DT|DWORD|POINTER|REAL|LREAL|S5TIME|STRING|TIME|TIMER|TIME_OF_DAY|TOD|USINT|UINT|UDINT|ULINT|VOID|WORD|WSTRING|TON_TIME|TOF_TIME|TP_TIME)\\b",
```

Comment:
Good catch... TIA SCL is the worst. Anyway, they are all aliases of an "IEC_Timer" datatype. Add that as well while you're at it, please.
ref: https://liambee.me/siemens/ton-tof-tp-timers-and-different-use-cases/

Reply:
IEC_Timer added i last commit (693306028c61dadd00a83c2ef9ffcf1e3405b081)